core/domain/prod_validation_jobs_one_off.py

```diff
@@ -1870,6 +1874,463 @@ def _get_custom_validation_functions(cls):
+        }
+
+
+class TopicSnapshotContentModelValidator(BaseSnapshotContentModelValidator):
+    """Class for validating TopicSnapshotContentModel."""
+
+    related_model_name = 'topic'
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'topic_ids': (
+                topic_models.TopicModel, [item.id[:item.id.find('-')]]),
+        }
+
+
+class TopicRightsModelValidator(BaseModelValidator):
+    """Class for validating TopicRightsModel."""
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        snapshot_model_ids = [
+            '%s-%d' % (item.id, version) for version in range(
+                1, item.version + 1)]
+        return {
+            'topic_ids': (
+                topic_models.TopicModel, [item.id]),
+            'manager_user_ids': (
+                user_models.UserSettingsModel, item.manager_ids),
+            'snapshot_metadata_ids': (
+                topic_models.TopicRightsSnapshotMetadataModel,
+                snapshot_model_ids),
+            'snapshot_content_ids': (
+                topic_models.TopicRightsSnapshotContentModel,
+                snapshot_model_ids),
+        }
+
+
+class TopicRightsSnapshotMetadataModelValidator(
+        BaseSnapshotMetadataModelValidator):
+    """Class for validating TopicRightsSnapshotMetadataModel."""
+
+    related_model_name = 'topic rights'
+
+    @classmethod
+    def _get_change_domain_class(cls, unused_item):
+        return topic_domain.TopicRightsChange
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'topic_rights_ids': (
+                topic_models.TopicRightsModel,
+                [item.id[:item.id.find('-')]]),
+            'committer_ids': (
+                user_models.UserSettingsModel, [item.committer_id])
+        }
+
+
+class TopicRightsSnapshotContentModelValidator(
+        BaseSnapshotContentModelValidator):
+    """Class for validating TopicRightsSnapshotContentModel."""
+
+    related_model_name = 'topic rights'
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'topic_rights_ids': (
+                topic_models.TopicRightsModel,
+                [item.id[:item.id.find('-')]]),
+        }
+
+
+class TopicCommitLogEntryModelValidator(BaseCommitLogEntryModelValidator):
+    """Class for validating TopicCommitLogEntryModel."""
+
+    related_model_name = 'topic'
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        # Valid id: [topic/rights]-[topic_id]-[topic_version].
+        regex_string = '^(topic|rights)-%s-\\d*$' % (
+            item.topic_id)
+
+        return regex_string
+
+    @classmethod
+    def _get_change_domain_class(cls, item):
+        if item.id.startswith('rights'):
+            return topic_domain.TopicRightsChange
+        else:
+            return topic_domain.TopicChange
```

Comment:
See similar comment in https://github.com/oppia/oppia/pull/6991#discussion_r296469646 -- I know you check it in _get_model_id_regex but it's far enough away from this function that it seems worth being careful anyway.

Reply:
Updated.